actionpack/lib/action_dispatch/middleware/remote_ip.rb

```diff
@@ -131,18 +131,26 @@ def initialize(env, middleware)
       # take the list of IPs, remove known and trusted proxies, and then take
       # the last address left, which was presumably set by one of those proxies.
       def calculate_ip
-        # Set by the Rack web server, this is a single value.
+        # We assume these things about the IP headers:
+        #
+        #   - X-Forwarded-For will be a list of IPs, one per proxy, or blank
+        #   - Client-Ip is propagated from the outermost proxy, or is blank
+        #   - either of the above could be a CSV list and/or repeated headers
+        #       that were concatenated.
+        #   - REMOTE_ADDR will be the IP that made the request to Rack
+
         remote_addr = ips_from('REMOTE_ADDR').last
```

Comment:
As of 4.2 the regex has been removed, I don't think there are any remaining perfomance considerations for this PR.